src/EA.Weee.Web/Areas/Admin/Mappings/ToViewModel/AatfDataToAatfDetailsViewModelMap.cs

```diff
@@ -45,7 +45,7 @@ public AatfDetailsViewModel Map(AatfDataToAatfDetailsViewModelMapTransfer source
 
             if (source.AssociatedAatfs != null)
             {
-                var associatedAEs = source.AssociatedAatfs.Where(a => a.FacilityType == FacilityType.Ae).ToList();
+                var associatedAEs = source.AssociatedAatfs.Where(a => a.FacilityType == FacilityType.Ae && a.Id != source.AatfData.Id).ToList();
```

Comment:
Can we create a unit test to test this as if our tests pass we are missing a unit test